AmplifyPlugins/API/AWSAPICategoryPlugin/Configuration/AWSAPICategoryPluginConfiguration+EndpointConfig.swift

```diff
@@ -281,13 +281,46 @@ public extension AWSAPICategoryPluginConfiguration {
 }
 
 extension Dictionary where Key == String, Value == AWSAPICategoryPluginConfiguration.EndpointConfig {
-    func getConfig(for apiName: String?,
-                   endpointType: AWSAPICategoryPluginEndpointType) throws ->
+
+    /// Getting the `EndpointConfig` resolves to the following rules:
+    /// 1. If `apiName` is specified, retrieve the endpoint configuration for this api
+    /// 2. If `apiName` is not specified, and `endpointType` is, retrieve the endpoint if there is only one.
+    /// 3. If nothing is specified, return the endpoint only if there is a single one, with GraphQL taking precedent
+    /// over REST.
+    func getConfig(for apiName: String? = nil,
+                   endpointType: AWSAPICategoryPluginEndpointType? = nil) throws ->
         AWSAPICategoryPluginConfiguration.EndpointConfig {
         if let apiName = apiName {
             return try getConfig(for: apiName)
         }
+        if let endpointType = endpointType {
+            return try getConfig(for: endpointType)
+        }
+
+        return try getConfig()
+    }
+
+    // Retrieve the endpoint configuration for the specified `apiName`
+    private func getConfig(for apiName: String) throws -> AWSAPICategoryPluginConfiguration.EndpointConfig {
+        guard let endpointConfig = self[apiName] else {
+
+            let error = APIError.invalidConfiguration(
+                "Unable to get an endpoint configuration for \(apiName)",
+                """
+                Review your API plugin configuration and ensure \(apiName) has a valid configuration.
+                """
+            )
+
+            throw error
+
+        }
 
+        return endpointConfig
+    }
+
+    // Retrieve the endpoint configuration when there is only one endpoint of the specified `endpointType`
+    private func getConfig(for endpointType: AWSAPICategoryPluginEndpointType) throws ->
+    AWSAPICategoryPluginConfiguration.EndpointConfig {
```

Comment:
nit: indentation. Did we run swiftFormat & swiftLint on this before submitting the PR?

Reply:
i believe it runs as part of build, but i can manually add indentation to the return type, and then increase the body's indentation by one more, then it will look like this:
```
private func getConfig(for endpointType: AWSAPICategoryPluginEndpointType) throws ->
        AWSAPICategoryPluginConfiguration.EndpointConfig {
            let apiForEndpointType = filter { (_, endpointConfig) -> Bool in
  
```